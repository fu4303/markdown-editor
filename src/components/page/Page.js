import React from "react"
import PropTypes from "prop-types"
import { PageContent } from "./styles"

const Page = ({ children, title }) => (
  <PageContent>
    <h2>{title}</h2>
    <hr />
    {children}
  </PageContent>
)

Page.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
}

export default Page
