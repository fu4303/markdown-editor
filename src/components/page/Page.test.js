import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import Page from "./Page"

describe("<Button />", () => {
  it("renders without crashing", () => {
    shallow(
      <Page title="page prop title">
        <p>Page children content</p>
      </Page>
    )
  })

  it("matches snapshot with data passed correctly", () => {
    const wrapper = shallow(
      <Page title="page prop title">
        <p>Page children content</p>
      </Page>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
