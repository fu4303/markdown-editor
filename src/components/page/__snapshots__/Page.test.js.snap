// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot with data passed correctly 1`] = `
<Fade
  delay={250}
  duration={1500}
>
  <Styled(main)>
    <h2>
      page prop title
    </h2>
    <hr />
    <p>
      Page children content
    </p>
  </Styled(main)>
</Fade>
`;
